src/main/kotlin/com/openlattice/search/requests/EntityNeighborsFilter.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.search.requests
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.openlattice.client.serialization.SerializationConstants
+import java.util.UUID
+import java.util.Optional
+
+data class EntityNeighborsFilter(
+        @JsonProperty(SerializationConstants.ENTITY_KEY_IDS) val entityKeyIds: Map<UUID, Set<UUID>>,
```

Comment:
this change is going to break a LOT of things, since basically all the apps use this in a lot of places and expect to pass a Set<UUID> -- for purposes of apps/services continuing to work, it would be good to continue support for the old API and just deprecate it, so we can move everything over to using the new format.

Reply:
Yeah, I thought that is going to affect a lot of things, it just made much more sense to keep it like this for searches with multiple entity sets, since all linked searches now require to have a clear mapping of linked entity sets and linking ids. 
I will re-create the old class and API endpoints to support the old version

Reply:
ok -- yeah if we're supporting neighbors searches across multiple entity sets that definitely does make more sense, it's just that it'll be super hard to coordinate re-releasing all the frontend projects and microservices at the exact time that this is deployed, since they really all will need to change in substantial ways.

Reply:
Re-added the old endpoints with deprecated flags